bundles/statistics/statsgrid2016/instance.js

```diff
@@ -318,23 +324,61 @@ Oskari.clazz.define(
             }
             return state;
         },
-        showToggleButtons: function(enabled, visible) {
-            var me = this;
-            var sandbox = me.getSandbox();
+        getEmbeddedTools: function () {
+            return this.embeddedTools;
+        },
+        /**
+         * Check if the tool has been added to the array containing tool names opened in publisher
+         * @method publisherHasTool
+         * @param {String} tool tool name as string
+         */
+        publisherHasTool: function (tool) {
+            return this.embeddedTools.indexOf(tool) > -1;
+        },
+        /**
+         * toggles the specified flyout from publisher
+         * @method toggleEmbeddedTools
+         * @param {String} tool tool name as string
+         */
+        toggleEmbeddedTools: function (tool) {
+            if ( this.publisherHasTool(tool) ) {
+                this.getTile().toggleFlyout( tool );
+            }
+        },
+        /**
+         * toggles the specified flyout from publisher
+         * @method showEmbeddedTools
+         * @param {String} tool tool name as string
+         * @param {boolean} enabled is the tool enabled or not
+         */
+        showEmbeddedTools: function (tool, enabled) {
+            var sandbox = this.getSandbox();
             var mapModule = sandbox.findRegisteredModuleInstance('MainMapModule');
-            if(!enabled) {
-                if(this.togglePlugin) {
-                    mapModule.unregisterPlugin(me.togglePlugin);
-                    mapModule.stopPlugin(me.togglePlugin);
+
+            if ( !enabled ) {
```

Comment:
This has so many branches that it's difficult to follow. Maybe add some comments at least. Which branch do you expect to activate, when and why.